Formula/testkube.rb

```diff
@@ -0,0 +1,36 @@
+class Testkube < Formula
+  desc "☸️ Kubernetes-native framework for test definition and execution"
+  homepage "https://testkube.io"
+  url "https://github.com/kubeshop/testkube/archive/refs/tags/v0.10.1.tar.gz"
+  sha256 "ddcc802c4150bf456dcc3c4b0a3323675ad41e4b200908d4eec23aadb27063d1"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    user = "brew"
+    tracking_id = "UA-204665550-8"
+    ENV["CGO_ENABLED"] = "0"
+    ldflags = %W[
+      -s -w
+      -X main.version=#{version}
+      -X main.builtBy=#{user}
+      -X github.com/kubeshop/testkube/pkg/analytics.testkubeTrackingID=#{tracking_id}
+    ]
+    (buildpath/"src/github.com/kubeshop/testkube").install buildpath.children
+    ENV.append_path "PATH", buildpath/"bin"
+
+    cd "src/github.com/kubeshop/testkube" do
+      system "go", "build", *std_go_args(output: "kubectl-testkube", ldflags: ldflags), "cmd/kubectl-testkube/main.go"
+      bin.install "kubectl-testkube"
+    end
+  end
+
+  test do
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
In general, this is an api client, that calls rest api server running in kubernetes. It needs kube config in order to connect kubernetes cluster. So, every command except `help` tries to reach kubernetes api. Even `version` command, because it checks both client and server versions.  I beleive first test is a response for
```If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.```

Reply:
Well, "Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook."

They usually trigger a different code path so let's just just run a command that a normal user would and check for the error there.